src/sentry/runner/initializer.py

```diff
@@ -34,7 +34,18 @@ def register_plugins(settings):
             plugins.register(plugin)
 
     for plugin in plugins.all(version=None):
-        plugin.setup(bindings)
+        init_plugin(plugin)
+
+
+def init_plugin(plugin):
+    from sentry.plugins import bindings
+    plugin.setup(bindings)
+
+    # Register contexts from plugins if necessary
+    if hasattr(plugin, 'get_custom_contexts'):
```

Comment:
I think this is good enough. It's just because v1 plugins also come through here :(